cmd/proxy/Dockerfile

```diff
@@ -22,9 +22,9 @@ COPY --from=builder /usr/local/go/bin/go /bin/go
 
 # Add tini, see https://github.com/gomods/athens/issues/1155 for details.
 RUN apk add --update bzr git mercurial openssh-client subversion procps fossil tini && \
-	mkdir -p /usr/local/go
-
-ENV GO_ENV=production
+	mkdir -p /usr/local/go && \
+	sed -i -r -e 's/(GoEnv\s*=\s*)"development"/\1"production"/' /config/config.toml && \
```

Comment:
That looks cray. Do you mind explaining what it does (i'm not a bash expert). 

But in any case, I think we can just remove it and have development be be the new default coming from the config. It's a soft breaking change but I can't imagine it actually breaking people. We're not 1.0 so that to me seems like an okay change.

Reply:
Hey! Sures, a quick rundown:
* `-i` : Edit the file in place (don't create a backup nor send to stdou)
* `-r` : Enable extended regex features, in this case back-referencing (see `\1`)
* `-e` : Prelude to an expression, i.e the proceeding `"/s..."`
* `s/find/replace/`: Format for wrapping a substitution expression.
* `(GoEnv\s*=\s*)"development"` : Look for the `GoEnv = "development"` property assignment.
* `(Go...)`: Parens tell the script to remember what it found, as we might use it later in the replace.
* `\s*=\s*` : Allow whitespace to be optional before and after the `=` i.e. could match `GoEnv="development"` as well as `GoEnv = "development"`
* `\1` : Take the text we found in the earlier `()` and stick it here
* `/config.config.toml` : Target file to look in (course)
* `&& \` : After executing this replacement, continue to execute statements on the following line(s).

re: `I think we can just remove it and have development be be the new default`

I'm happy to submit another PR for that if you're sure its route you want to take - This PR (And the other) were just in case you were hold-fast on having production mode enable in the docker hub images.

-DF

Reply:
If simply removing the statement is deemed viable, then I have created PR #1443 to accomplish that.

BTW: Thank you for reviewing my PRs !